### PR TITLE
Adds Output Field Argument Types Mergable to Composition

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

File: spec/Section 4 -- Composition.md [hunk 1]
```diff
@@ -14,6 +14,96 @@ run in sequence to produce the composite execution schema.
...
+
+**Error Code**
+
+OUTPUT_FIELD_ARGUMENT_TYPES_NOT_MERGEABLE
+
+**Formal Specification**
+
+- Let {fieldsByName} be a map of field lists where the key is the name of a
+  field and the value is a list of fields from mergeable types from different
+  subgraphs with the same name.
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:27]: Source schemas?

File: spec/Section 4 -- Composition.md [hunk 2]
```diff
@@ -14,6 +14,96 @@ run in sequence to produce the composite execution schema.
...
+- for each {fields} in {fieldsByName}
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:28]:
```suggestion
- For each {fields} in {fieldsByName}
```

File: spec/Section 4 -- Composition.md [hunk 3]
```diff
@@ -14,6 +14,96 @@ run in sequence to produce the composite execution schema.
...
+- Given each pair of arguments {argumentA} and {argumentB} in {fieldA} and
+  {fieldB}:
+  - Let {typeA} be the type of {argumentA}
+  - Let {typeB} be the type of {argumentB}
+  - {SameInputTypeShape(typeA, typeB)} must be true.
+
+**Explanatory Text**
+
+Fields on mergeable objects or interfaces that have the same name are considered semantically 
+equivalent and mergeable when they have a mergeable argument types.
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:47]:
```suggestion
equivalent and mergeable when they have mergeable argument types.
```

**Review by glen-84 — CHANGES_REQUESTED** [review]

**Comment by glen-84** [comment]
PR title:

```diff
-Adds Output Field Argument Types Mergable to Composition
+Adds Output Field Argument Types Mergeable to Composition
```

File: spec/Section 4 -- Composition.md [hunk 4]
```diff
@@ -14,6 +14,103 @@ run in sequence to produce the composite execution schema.
...
+
+**Severity**
+
+ERROR
+
+**Formal Specification**
+
+- Let {typeNames} be the set of all output type names from all source schemas.
+- For each {typeName} in {typeNames}
+  - Let {types} be the set of all types with the name {typeName} from all source
```
Comment (michaelstaib) [inline spec/Section 4 -- Composition.md:31]:
```suggestion
  - Let {types} be the set of all types with the {typeName} from all source
```

**Comment by PascalSenn** [comment]
Closed by #93

**Pull request closed without merging**